From 0d8f6dbfe1035d74c0bd9149f1a3ae4dac5aeae4 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.de>
Date: Fri, 6 Mar 2015 11:46:41 +0100
Subject: [PATCH] Do not switch internal CDC device on IBM NeXtScale nx360 M5
Git-Commit: 0d8f6dbfe1035d74c0bd9149f1a3ae4dac5aeae4
Patch-Mainline: Never (made moot by later feature not taken in SP3)
References: bnc#913598

On SP3 we don't do ring expansion. Firmware upload on the
internal CDC device of these systems overflows the ring.
Thus we leave them on EHCI.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
---
 drivers/usb/host/pci-quirks.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/usb/host/pci-quirks.c b/drivers/usb/host/pci-quirks.c
index 0c9f08b..b0227da 100644
--- a/drivers/usb/host/pci-quirks.c
+++ b/drivers/usb/host/pci-quirks.c
@@ -606,6 +606,13 @@ static const struct dmi_system_id bt_excepted_from_switch_table[] = {
                 },
                 .driver_data = (void *)64,
         },
+	{
+		/* IBM NeXtScale nx360 M5 */
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_NAME, "IBM NeXtScale nx360 M5"),
+		},
+		.driver_data = (void *)1,
+	},
 	{ }
 };
 
-- 
2.1.4

